chromadb/test/property/test_embeddings.py

```diff
@@ -173,6 +175,10 @@ def _upsert_embeddings(self, embeddings: strategies.RecordSet):
                     self.embeddings["embeddings"][target_idx] = embeddings[
                         "embeddings"
                     ][idx]
+                else:
+                    self.embeddings["embeddings"][target_idx] = self.embedding_function(
```

Comment:
This is very ugly, and IMO this entire code block should be refactored. It's very difficult to read and reason about.